datasetUtils/checkForNewVersion_test.go

```diff
@@ -0,0 +1,102 @@
+package datasetUtils
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"bytes"
+	"log"
+)
+
+func TestFetchLatestVersion(t *testing.T) {
+	// Create a mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Return a JSON response similar to the GitHub API
+		w.Write([]byte(`{"tag_name": "v1.0.0"}`))
+	}))
+	defer server.Close()
+	
+	// Replace GitHubAPI with the URL of the mock server
+	oldGitHubAPI := GitHubAPI
+	GitHubAPI = server.URL
+	defer func() { GitHubAPI = oldGitHubAPI }()
+	
+	// Create a mock HTTP client
+	client := server.Client()
+	
+	// Call fetchLatestVersion
+	version, err := fetchLatestVersion(client)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	
+	// Check the version number
+	if version != "v1.0.0" {
+		t.Errorf("Expected version v1.0.0, got %s", version)
+	}
+}
+
+func TestCheckForNewVersion(t *testing.T) {
+	tests := []struct {
+		name           string
+		currentVersion string
+		mockResponse   string
+		expectedLog    string
+		}{
+			{
+				name:           "New version available",
+				currentVersion: "0.9.0",
+				mockResponse:   `{"tag_name": "1.0.0"}`,
+				expectedLog:    "You should upgrade to a newer version",
+			},
+			{
+				name:           "No new version available",
+				currentVersion: "1.0.0",
+				mockResponse:   `{"tag_name": "1.0.0"}`,
+				expectedLog:    "Your version of this program is up-to-date",
+			},
+		}
+		
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a mock HTTP server
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// Return a JSON response similar to the GitHub API
+				w.Write([]byte(tt.mockResponse))
+			}))
+			defer server.Close()
+			
+			// Replace GitHubAPI with the URL of the mock server
+			oldGitHubAPI := GitHubAPI
+			GitHubAPI = server.URL
+			defer func() { GitHubAPI = oldGitHubAPI }()
+			
+			// Create a mock HTTP client
+			client := server.Client()
+			
+			// Call CheckForNewVersion
+			CheckForNewVersion(client, "test", tt.currentVersion, false)
+			
+			// Check the log output
+			logOutput := getLogOutput()
+			if !strings.Contains(logOutput, tt.expectedLog) {
+				t.Errorf("Expected log message not found: %s", logOutput)
+			}
+			
+			// Clear the log buffer after each test
+			buf.Reset()
```

Comment:
this could maybe be in a cleanup? https://pkg.go.dev/testing#B.Cleanup

Reply:
Maybe, but this is also one liner. Does the job good enough.

Reply:
ok. Even though it is a little weird that the setup is in the init while the teardown is in the function

Reply:
I could move 
```
    var buf bytes.Buffer
    log.SetOutput(&buf)
```
inside the `TestCheckForNewVersion` function. But if we were to add new tests in the future, we'd have to rewrite that.

Reply:
logically the "afterTest" part should belong to the cleanUp, so I'd rather move that there, rather than moving the init inside the func. But if you prefer having it here, that is fine with me

Reply:
I don't have any personal preference here, other than avoiding redundancy. I just don't see a way to avoid having the clean up inside the test function. If we have multiple tests, we want them to clean up before the next test.

Reply:
that's what the cleanup should do as the init does, it's a builtin function of the go test module if I understand right, that gets automatically called after each test. But I am fine with this anyway (also because I am not sure it does this in go)!

Reply:
Yeah but I think the `t.Cleanup()` method still needs to be called from within the test function. It can't be called from outside the test function, because it needs a reference to the `testing.T` instance for the current test.